src/audiovideocontroller/DefaultAudioVideoController.ts

```diff
@@ -492,6 +492,50 @@ export default class DefaultAudioVideoController
     );
   }
 
+  async replaceLocalVideo(): Promise<void> {
+    let videoStream: MediaStream | null = null;
+    try {
+      videoStream = await this.mediaStreamBroker.acquireVideoInputStream();
+    } catch (error) {
+      this.logger.info('could not acquire video stream from mediaStreamBroker');
+    }
+    if (!videoStream || videoStream.getVideoTracks().length < 1) {
+      throw new Error('could not acquire video track');
+    }
+
+    const videoTrack = videoStream.getVideoTracks()[0];
+    if (!this.meetingSessionContext || !this.meetingSessionContext.peer) {
+      throw new Error('no active meeting and peer connection');
+    }
+
+    if (this.meetingSessionContext.browserBehavior.requiresUnifiedPlan()) {
+      await this.meetingSessionContext.transceiverController.setVideoInput(videoTrack);
+    } else {
+      throw new Error('cannot replace track on Plan B');
+    }
+
+    // if there is a local tile, a video tile update event should be fired.
+    const localTile = this.meetingSessionContext.videoTileController.getLocalVideoTile();
+    if (localTile) {
+      const state = localTile.state();
+      const settings = videoStream.getVideoTracks()[0].getSettings();
+      // so tile update wil be fired.
+      localTile.bindVideoStream(
+        state.boundAttendeeId,
+        true,
+        videoStream,
+        settings.width,
+        settings.height,
+        state.streamId,
+        state.boundExternalUserId
+      );
+    }
+
+    // subscription context should be updated to reflect the change so it may be reused and destroyed
+    // in subsequent meeting action.
```

Comment:
Is this a TODO, or a statement of what the next line is doing?

Reply:
statement explaining the next line.

Reply:
Suggest more active phrasing:

```
// Update subscription context to match what we just changed
// so that subsequent meeting actions can reuse and destroy it.
```

Reply:
updated.